src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -720,6 +721,16 @@ Status ServiceBasedNodeInfoAccessor::AsyncReportHeartbeat(
   return Status::OK();
 }
 
+void ServiceBasedNodeInfoAccessor::AsyncReReportHeartbeat(
+    bool is_pubsub_server_restarted) {
+  if (!is_pubsub_server_restarted && cached_heartbeat_.has_heartbeat()) {
```

Comment:
Isn't is supposed to be this predicate based on your description? 

```
if (is_pubsub_server_restarted && ...)
```

Reply:
Nah..We only resend heartbeat when gcs server restarts, but not care pubsub server's status. In codes `is_pubsub_server_restarted` is always false as we do not support the true condition for now.

I don't know if there will exist the scenario that only pubsub server restarts with gcs server staying healthy (now redis failure causes gcs server failure), so I would like just remove this flag check as resend once more heartbeat would not  impact much.